docs/tracing/docs/030-tracing-showcase.md

```diff
@@ -0,0 +1,148 @@
+
+![](assets/showcase-only-publish-overview.png)
+
+In the trace details, you can see the tags for the "publish-service"
+
+![](assets/showcase-only-publish-detail.png)
+
+---
+
+### I have a micro-service / lambda Event trigger configured
+
+In such case, you will also see the "webhook-service" in the trace along with the "publish-service". There will be more spans from Envoy. You can also see the spans from the lambda or service for a downstream call if the tracing context headers are propagated while making those calls.
+
+For example: there is the following flow in a given trace:
+
+ec event → kyma gateway (nginx) → publish-service → webhook → push (envoy) → service/lambda (envoy) → call to ec → http-db-service
+
+![](assets/showcase-with-microservice-overview.png)
+
+The journey would look like the call propagation:
+
+![](assets/showcase-with-microservice-detail1.png)
+
+The details in the "webhook-service" provides information about the endpoint to which Events are delivered.
+
+![](assets/showcase-with-microservice-detail2.png)
+
+---
+
+### I have an error
+In case there is an error, the corresponding span is marked as error.
+
+For example, in the following span, the service receiving the Event fails due to the code bug:
+
+![](assets/showcase-error-overview.png)
+
+If you look at the "webhook-service", you can see more error details. The service sent back "HTTP 500" when the Event Bus attempted to deliver the Event.
+
+![](assets/showcase-error-detail1.png)
+
+Since the Event Bus keeps on retrying to deliver the Event until it is successful, you can see multiple spans for the "webhook-service":
+
+![](assets/showcase-error-detail2.png)
+
+---
+
+### I want to search by tags
+
+It is possible to search the traces using tags. Tags are "key-value" pairs configured for each service.
+
+Here are some examples:
+
+* Search by "event-type"
+
+![](assets/showcase-search-by-tag-event-type.png)
+
+* Search by "event-type" and "event-id"
+
+![](assets/showcase-search-by-tag-event-type-and-event-id.png)
+
+* Search for error:
+
+  In this case, you must choose the service which has the error and then use the tag.
+
+![](assets/showcase-search-by-tag-error.png)
+
+---
+
+### I want to ensure tracing for the downstream service calls
+
+Ensure tracing for the downstream service calls when you create a lambda or a microservice which is called by another lambda or a microservice triggered by the Events. For example, a lambda called "test-lambda" triggered by the "order.created" Event calls an "http-db-service" microservice to store the order details.
+
+Do the following to ensure that the spans are shown in the trace:
+1.	Pass the trace context headers. Refer to this documentation.
+2.	Add the label to "app:{name of lambda or service}" when creating the lambda or service.
+3.	Name the port "http" or with the prefix "http-" (reference: https://istio.io/help/faq/traffic-management/#naming-port-convention).
```

Comment:
Do we need a label for both service and deployment?
Is it possible to somehow highlight the relevant parts of the code sample?